src/__tests__/not-found.js

```diff
@@ -0,0 +1,26 @@
+import '@testing-library/jest-dom'
```

Comment:
**só um comentário**, acho interessante a escolha de ter o teste aqui numa pasta a parte so pra tests. Eu tenho visto recentemente a pratica de ter o teste junto do componente. Por exemplo, na mesma pasta do `index.js` do NotFound, ter `index.test.js`. Mas percebo que ambos devem funcionar, é só uma questão de saber o standard que queremos usar no projeto

Reply:
Tbh, é a primeira vez que faço testes unitários. É justamente um ponto que quero explorar com maior profundidade porque ainda não tive essa oportunidade profissionalmente. A forma como vi na documentação do testing-library foi com a pasta para tests, mas também já vi bastante como você disse, de forma individualizada, dentro dos componentes. 

Talvez uma sugestão seja colocar os testes unitários localmente com os componentes e deixar a pasta de testes para outros mais extensos, como end-to-end ou de integração, caso façam. 

Para já, é como preferirem :)

Reply:
Muito fixe :) Fico feliz do projeto estar a dar essa oportunidade.
Eu por acaso tentei escrever um para ver o quão fazível era, e acabei por não completar e deixar para outros contribuidores se aventurarem 😅 Fizeste um bom trabalho!

Tenho curiosidade das reviews que tiveres aqui, certamente também vou aprender com esta PR :)

Reply:
> **só um comentário**, acho interessante a escolha de ter o teste aqui numa pasta a parte so pra tests. Eu tenho visto recentemente a pratica de ter o teste junto do componente. Por exemplo, na mesma pasta do `index.js` do NotFound, ter `index.test.js`. Mas percebo que ambos devem funcionar, é só uma questão de saber o standard que queremos usar no projeto

Há diversas formas, na maioria vi dentro de cada pasta __tests__ com os testes lá dentro correspondente aos ficheiros de cada pasta

Reply:
@juliamendesc obrigado por outra contribuição :) 

Esse caso de tests unitários acho muito interessante e tenho algumas ideias para essa task.

1.  Achei muito legal utilizar o approach que @toothlesspanda colocando tudo na pasta "__tests__". Eu adicionaria mais uma coisa: como está a trabalhar no página "Not found" colocaria o arquivo nessa caminho: "__tests__/pages/NotFound404/not-found.test.js" e com ".test." pra indicar ser um arquivo de tests.
2. Por se tratar de uma página stateless, o teste mais indicado acho que seria o de snapshot (https://jestjs.io/docs/snapshot-testing). Dessa forma consegue controlar o que é renderizado na DOM como código, ou seja, se o conteúdo do snapshot mudar implica que a UI foi alterada :)
3. Acho que deveria existir um arquivo um arquivo "setupTests.js" para conseguir fazer o setup dos tests em um único lugar. (https://create-react-app.dev/docs/running-tests/#srcsetuptestsjs)

Vou comentar algumas coisas no seu código :)

Reply:
Pra fazer essa importação: 
import '@testing-library/jest-dom' cria um arquivo dentro de src com o nome "setupTests.js" e cola essa importação la dentro, por favor. Assim a cada novo teste criado não é necessário importar novamente.

Feito isso vai ter que ir na raiz do projeto e adicionar esse código:
  "setupFilesAfterEnv": ["<rootDir>/src/setupTests.js"]
no objeto principal. Com isso o jest vai saber que sempre deve ir buscar as infos desse arquivo para fazer o setup dos tests :)

Reply:
@gustavocr19-developer fiz as alterações com algumas alterações. Fui ver a documentação do jest e lá a sugestão era um jest.setup.js e um jest.config.js. fiz algumas mudanças mais extensas e queria te pedir por favor, por favorzinho pra olhar o que mais posso melhorar 🫠